locator/tablets.hh

```diff
@@ -177,6 +177,9 @@ enum class tablet_transition_kind {
     // The leaving replica is (tablet_info::replicas - tablet_transition_info::next).
     migration,
 
+    // Like migration, but the new pending replica is on the same host as leaving replica.
+    intranode_migration,
+
```

Comment:
Please update docs/dev/topology-over-raft.md (the mermaid diagrams and maybe surrounding material).

Reply:
You mean the stage transition diagram? It's the same for "migration" and "intranode_migration" transitions.

Reply:
Added a note:
```
The above state transition state machine is the same for different tablet transition kinds: migration, intranode_migration, rebuild.

The behavioral difference between "migration" and "intranode_migration" transitions is in the way "streaming" stage
is performed. In case of intra-node migration, streaming is done by fast duplication of data by creating hard links to
sstable files on the destination shard. Original sstable files on the source shard will be removed by the standard "cleanup" stage.
```